modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2SynapseEnvironment.java

```diff
@@ -227,27 +227,20 @@ public void setSeverDebugMode(ServerContextInformation contextInformation) {
      * unit test message context and environment initializes
      */
     private void setSeverUnitTestMode(ServerContextInformation contextInformation) {
-        if (Boolean.parseBoolean(System.getenv("UNIT_TEST"))) {
+        if (Boolean.parseBoolean(System.getProperty("synapseTest"))) {
             setUnitTestEnabled(true);
             contextInformation.setServerUnitTestModeEnabled(true);
-            log.info("Synapse unit testing enabled");
-
-            //get unit test received port
-            String requestPort = System.getenv("UNIT_TEST_PORT");
-
-            //set default port for TCP server connection
-            int unitTestingAgentPort = 9007;
-            if (requestPort != null && !requestPort.isEmpty()) {
-                try {
-                    unitTestingAgentPort = Integer.parseInt(requestPort);
-                } catch (NumberFormatException e) {
-                    log.error("Given port is not in valid format", e);
-                }
+            log.info("Synapse unit testing server enabled");
+
+            //check unit test received port, if its null then set default port as 9008
+            String requestPort = System.getProperty("synapseTestPort");
```

Comment:
I think we can move this inside UnitTestingExecutor, WDYT?

Reply:
Yes, we can remove this from here. Changed it as suggested.